app/actions/handlers.py

```diff
@@ -0,0 +1,151 @@
+import httpx
+import logging
+
+import app.actions.client as client
+
+from datetime import datetime, timedelta, timezone
+from app.actions.configurations import AuthenticateConfig, PullObservationsConfig, PullFarmObservationsConfig, get_auth_config
+from app.services.action_scheduler import trigger_action
+from app.services.activity_logger import activity_logger
+from app.services.gundi import send_observations_to_gundi
+from app.services.state import IntegrationStateManager
+from app.services.utils import generate_batches
+
+logger = logging.getLogger(__name__)
+state_manager = IntegrationStateManager()
+
+
+RUMI_BASE_URL = "https://innogando-backend-prod-01.innogando.com"
+
+
+def transform(farm, observation):
+    lat_lon = observation.location.split("::")
```

Comment:
This might be more readable if we do this. And it should protect against a change in the format for `location`.
```suggestion
    lat, lon = observation.location.split("::", maxsplit=2)
```

You might also consider putting this in the pydantic model -- keeping the validation all in one place.